library_management_system_downloader/run_api_download_and_parse_results.py

```diff
@@ -0,0 +1,201 @@
+
+# =============================================================================
+# Load configuration variables
+# =============================================================================
+
+# Location of configuration file.
+# This should be a python script (named with a .py file extension), and
+# should contain  variables named the following:
+# - static_parameters_for_api (a dictionary)
+# - user_agent_custom_string (a string)
+# - api_base_url (a string)
+# -tsv_dataset_location (a filepath string)
+
+from library_management_system_downloader \
+        import downloader_configuration_file as config
+# config.api_base_url
+# config.user_agent_custom_string
+# config.static_parameters_for_api
+# config.tsv_dataset_location
+
+from library_management_system_downloader import create_api_request as api
+from library_management_system_downloader \
+        import evaluate_api_response_for_fulltext_indication as fulltext
+
+# =============================================================================
+# Create additional settings
+# =============================================================================
+
+logging.basicConfig(level=logging.INFO)
+
+# =============================================================================
+# Set up a connection to (and tables within) an SQLite database:
+# =============================================================================
+
+# Initialize the database:
+sql_engine = create_engine(
+        'sqlite:///library_coverage_xml_and_fulltext_indicators.db')
+
+
+# From the sqlalchemy documentation, at
+# http://docs.sqlalchemy.org/en/latest/dialects/sqlite.html:
+# Turn on write-ahead logging, which will allow reading and writing at
+# the same time (without this, I was getting an error re: the database being
+# locked when chunksize is used with read_sql_query, since that returns a
+# generator and is thus still reading the database:
+@event.listens_for(sql_engine, "connect")
+def set_sqlite_pragma(sql_engine, connection_record):
+    cursor = sql_engine.cursor()
+    cursor.execute("PRAGMA journal_mode=WAL")
+    cursor.close()
+
+# Create an sqlalchemy metadata schema:
+metadata = MetaData(sql_engine)
+
+# Create a schema for the table that will hold the API data:
+# The full_text_indicator column will be an integer in that 1 = fulltext
+# available and 0 = fulltext not available:
+dois_table = Table(
+        'dois_table', metadata,
+        Column('database_id', Integer, primary_key=True),
+        Column('doi', String, nullable=False, unique=True, index=True))
+
+library_holdings_table = Table(
+        'library_holdings_data', metadata,
+        Column('doi_foreign_key',
+               String, ForeignKey("dois_table.database_id"), nullable=False),
+        Column('timestamp', DateTime, nullable=False,
+               default=func.now()),  # Default is creation time
+        Column('xml_response', String, nullable=False),
+        Column('full_text_indicator', Integer))
```

Comment:
I'm not sure `full_text_indicator` should be part of the database as opposed to created during a post-processing step. My thought was that the database would be solely for storing raw responses and avoiding duplicate API queries. The results could be processed later.

I don't have a super strong feeling here. If you go this way, you could save all XML processing for a future PR and we could just focus on API queries in this one.

Reply:
I see the argument both ways. My thought here is that the XML processing is minimal -- the only thing being done with the XML is to make a binary decision about whether full-text is available. Thus, since the `full_text_indicator` column just reflects what's in the XML, and not a big transformation of the XML, it makes sense to me to just store it alongside. Having said that, I also don't have super strong feelings about this, and the processing step is already its own function.